### PR TITLE
[improve] [log] Print source client addr when enabled haProxyProtocolEnabled

### PR DESCRIPTION
**Pull request by poorbarcode**
### Motivation

`PulsarHandler.remoteAddress` is the socket address of the Proxy when enabled `haProxyProtocolEnabled`, we'd better print three addresses: `local address`, `remote address`, `source remote address` in the log.

### Modifications
Print three addresses: `local address`, `remote address`, `source remote address` in the log.

After this PR, the demo of addresses in the log is below:
- `Demo: [id: 0x2561bcd1, L:/10.0.136.103:6650! R:/240.240.0.5:58038] [SR:/240.240.0.5:58038]`
- `L`: local Address
- `R`: remote address
- `SR`: source remote address. It is the source address when enabled "haProxyProtocolEnabled".

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository: x

### CODE REVIEW

**Review by Technoboy- — APPROVED** [review]

File: pulsar-common/src/main/java/org/apache/pulsar/common/protocol/PulsarHandler.java [hunk 1]
```diff
@@ -152,5 +152,19 @@ public void cancelKeepAliveTask() {
...
+     * Demo: [id: 0x2561bcd1, L:/10.0.136.103:58038 ! R:/240.240.0.5:6650].
+     * L: local Address.
+     * R: remote address.
+     */
+    @Override
+    public String toString() {
+        if (ctx == null) {
+            return "[ctx: null]";
+        } else {
+            return ctx.channel().toString();
```
Comment (lhotari) [inline pulsar-common/src/main/java/org/apache/pulsar/common/protocol/PulsarHandler.java:165]: Is channel always non-null?

**Review by shibd — APPROVED** [review]

File: pulsar-common/src/main/java/org/apache/pulsar/common/protocol/PulsarHandler.java (hunk 1 above)
Comment (poorbarcode) [reply pulsar-common/src/main/java/org/apache/pulsar/common/protocol/PulsarHandler.java:165]: Yes, in normal the `ctx` is typed `AbstractChannelHandlerContext`, and the variable `pipeline` with modifier `final` will be initialized in the constructor, and the variable `channel` of the pipeline is also initialized in its constructor.

**Review by lhotari — APPROVED** [review]
LGTM

**Pull request merged**